pkg/runner/local_docker.go

```diff
@@ -391,6 +429,14 @@ func (r *LocalDockerRunner) Run(ctx context.Context, input *api.RunInput, ow *rp
 	ctxContainers, cancel := context.WithCancel(ctx)
 	defer cancel()
 
+	eventsWg := sync.WaitGroup{}
+	eventsCancel, err := r.collectOutcomes(ctxContainers, result, &eventsWg, &template)
```

Comment:
I am a bit confused with the flow here. Why are we passing a WaitGroup to `collectOutcomes` and waiting for it only in the `select` below? I don't understand the changed flow here... also the WaitGroup has a single entry, so I am not sure of its purpose.

Reply:
I am using the flow of `SubscribeEvents` instead of fetching them all in the end as described here: https://github.com/testground/testground/issues/1160. Maybe a channel is better to indicate that it finished instead of a waiting group.

If you think it's better to just collect them in the end as we did before, please let me know.